server.js

```diff
@@ -1,28 +1,127 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
+import express from "express"
+import bodyParser from "body-parser"
+import cors from "cors"
+import mongoose from "mongoose"
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  name: {
+    type: String,
+    default: "Anonymous"
+  },
+  theme: {
+    type: String,
+    default: "❤️",
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date(),
+  },
+})
+
+const port = process.env.PORT || 2500
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+
+const myEndpoints = require("express-list-endpoints")
+app.get("/", (req, res) => {
+  res.send(myEndpoints(app))
 })
 
+//Handling erorrs by summing them in consts
+const COULD_NOT_SAVE_THOUGHT = "Sorry! Could not save thought to the database";
+const COULD_NOT_FIND_THOUGHT_WITH_ID = "Sorry! Unable to find the thought with the ID: ";
+const COULD_FIND_THOUGHT = "Added one more like";
+
+//GET THOUGHTS
+app.get("/thoughts", async (req, res) => {
+  const { page, sort } = req.query;
+
+  //Pagination to display 20 thoughts per page and go onto skip page once the count is fullfilled
+  const pageNo = +page || 1;
+  const thoughtsPerPage = 20;
+  const skip = thoughtsPerPage * (pageNo - 1);
+
+  const allThoughts = await Thought.find();
+  const pages = Math.ceil(allThoughts.length / thoughtsPerPage);
+
+  //Sort thoughts by the ones created most recently / returns the amount of thoughts and pages
+  const sortAllThoughts = (sort) => {
+    if (sort === "likes") {
+      return {
+        hearts: -1,
+      };
+    } else if (sort === "oldest") {
+      return {
+        createdAt: 1,
+      };
+    } else {
+      return {
+        createdAt: -1,
+      };
+    }
+  };
+
+  const thoughts = await Thought.find()
+    .sort(sortAllThoughts(sort))
+    .limit(thoughtsPerPage)
+    .skip(skip)
+    .exec();
```

Comment:
Not sure what this code does as I don´t see you using this const later on in the code.